core/network/peer_state.hpp

```diff
@@ -95,6 +96,20 @@ namespace kagome::network {
       return fresh_implicit;
     }
 
+    /**
+     * Set of functions to manipulate in-parachain set of nodes.
+     */
+    bool can_be_disconnected() const {
+      return 0 == use_count;
+    }
+    void inc_use_count() {
+      ++use_count;
+    }
+    void dec_use_count() {
+      BOOST_ASSERT(use_count > 0);
+      --use_count;
+    }
```

Comment:
Use RAII object (e.g. `shared_ptr`) or use flags (e.g. `is_validator`, `is_group`, `is_grid`, ...) to avoid bugs with duplicate inc/dec.